### ISSUE TITLE
Button SEND

### ISSUE THREAD

**Issue by tavoguardp** [pluginsGLPI/formcreator#3116]
I have a form created.

I updated the plugin to the latest version but now after the plugin update, when filling the form, the submit button disappears preventing me to send the form.

thank you

Mi GLPI version is [GLPI 9.5.10](http://glpi-project.org/)
Form 
Version GPL v2+
TECLIB'
2.12.7

**Comment by btry** [comment]
Hi

You probably have an error when the plugin computes how to hide and show the button. 

Please open developer tools of the browser. Refresh the page then fill the form. In the network tab of the developer tools, you will find several requests to showfieds.php. check them. Their http response must be 200 and the response body must be a valid JSON string. 

Maybe the JSON is broken because of a PHP error. Do you have debug mode enabled ?

**Comment by tavoguardp** [comment]
Hi,

![image](https://user-images.githubusercontent.com/120193581/206751466-b7ff9ccf-e9f6-47c6-8c72-b245649cbaaf.png)
I performed the steps and still the button hides.
but it gives the same times as you say

**Comment by btry** [comment]
Show the response body of the latest http request to showfields.php

**Comment by tavoguardp** [comment]
> Muestra el cuerpo de respuesta de la última solicitud http a showfields.php

![image](https://user-images.githubusercontent.com/120193581/206752359-31bb82fc-ef74-43f0-9e50-819166047776.png)

**Comment by btry** [comment]
This is the request body (the data sent by the browser to the server). Click on the tab "Response" to show the response body (the data sent by the server to the browser).

**Comment by tavoguardp** [comment]
alt-svc: h3=":443"; ma=86400, h3-29=":443"; ma=86400
cache-control: no-store, no-cache, must-revalidate
cf-cache-status: DYNAMIC
cf-ray: 776f3b92fb058dd3-MIA
content-encoding: br
content-type: text/html; charset=UTF-8
date: Fri, 09 Dec 2022 16:49:12 GMT
expires: Mon, 26 Jul 1997 05:00:00 GMT
nel: {"success_fraction":0,"report_to":"cf-nel","max_age":604800}
pragma: no-cache
report-to: {"endpoints":[{"url":"https:\/\/a.nel.cloudflare.com\/report\/v3?s=GDJmnZUC0or2n8wr4WJaiJLJ0ia3ORwqVzUDcKFTDHO4sLU9sJ3yTjecLT4ZFEq55cUwOeOKG%2FsNGIDOAFA6vqLCLQndszLat07jD%2FD9cmp6CJwScxmTZtidR%2B6xb4cC6XLnKHmICrik%2Bw%3D%3D"}],"group":"cf-nel","max_age":604800}
server: cloudflare
strict-transport-security: max-age=63072000; includeSubDomains; preload
x-content-type-options: nosniff
x-frame-options: SAMEORIGIN
x-frame-options: SAMEORIGIN
x-powered-by: PHP/7.4.15
:authority: glpi.dcarosolutions.com
:method: POST
:path: /marketplace/formcreator/ajax/showfields.php
:scheme: https
accept: */*
accept-encoding: gzip, deflate, br
accept-language: es-ES,es;q=0.9
content-length: 451
content-type: application/x-www-form-urlencoded; charset=UTF-8
cookie: glpi_3f946f74140a3178722cb675d5bf6b47=c2b6g956dhanso2ctgfperaif5
origin: https://glpi.dcarosolutions.com
referer: https://glpi.dcarosolutions.com/marketplace/formcreator/front/formdisplay.php?id=1
sec-ch-ua: "Not?A_Brand";v="8", "Chromium";v="108", "Google Chrome";v="108"
sec-ch-ua-mobile: ?0
sec-ch-ua-platform: "Windows"
sec-fetch-dest: empty
sec-fetch-mode: cors
sec-fetch-site: same-origin
user-agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/108.0.0.0 Safari/537.36
x-glpi-csrf-token: b805050632b7b8fa456a6510ab9e7bf59eba60e31bff1f0847c09ca5622646e9
x-requested-with: XMLHttpRequest

**Comment by tavoguardp** [comment]
> This is the request body (the data sent by the browser to the server). Click on the tab "Response" to show the response body (the data sent by the server to the browser).

Can you help me?

**Comment by btry** [comment]
Hi

To show th response body of the request, click on the "response" tab

![image](https://user-images.githubusercontent.com/14139801/206989541-96541c2d-cb72-4c62-8c83-f0e26d2d7d86.png)

**Comment by tavoguardp** [comment]
> Hi
> 
> To show th response body of the request, click on the "response" tab
> 
> ![image](https://user-images.githubusercontent.com/14139801/206989541-96541c2d-cb72-4c62-8c83-f0e26d2d7d86.png)

Hi

"{"PluginFormcreatorQuestion":{"1":true,"9":true,"2":true,"3":true,"4":true,"5":true},"PluginFormcreatorSection":{"1":true,"2":true},"PluginFormcreatorForm":true}"

**Comment by btry** [comment]
It looks good. No error and the content says the button must show.

Other requests will happen whe you type something in a question. Add some input to the questions then show the state of the latest request. The button should be hidden, and the content of the http request should contant 1 different content

**Comment by tavoguardp** [comment]
I performed a test by upgrading to GLPI 10 and it works correctly.
I also performed the test of creating another new form and the same thing happens.
I also downloaded the version plugin and the same scenario also happened. 
It seems to be something in this version of GLPI 9.5.10.

**Comment by btry** [comment]
Hi

So upgrading to GLPI 10 solved your problem ?

**Comment by tavoguardp** [comment]
Yes, but since I am still working with fusion I can't get to that version yet.
I have it on a virtual machine.

**Comment by btry** [comment]
Hi

You may switch to GLPI Inventory: https://glpi-project.org/discover-native-glpi-inventory/

**Comment by tavoguardp** [comment]
If I upgrade to GLPI inventory will my merged agents still work?

**Comment by btry** [comment]
Hi

What is a **merged** agent ?

**Comment by tavoguardp** [comment]
Could you better tell me up to which version of glpi 10 still works with the fusion agent?

**Comment by btry** [comment]
All versions are compatible. I recommend you to always use the latest version of GLPI. When you have some time, consider to test then deploy GLPI Agent to ensure long term compatibility.

**Comment by btry** [comment]
Hi

Can I close this issue ?

**Comment by izaacmenezes** [comment]
![image](https://user-images.githubusercontent.com/72463556/210885633-4b139485-4a2f-4ce1-bc24-ff6ad3b9e874.png)


tive o mesmo erro na versão 9.5.1 do glpi conforme a imagem,

quando boto a pagina no modo desenvolvedor tenho o erro abaixo:

DevTools failed to load source map: Could not load content for http://"meu ip"/css_compiled/css_styles.min.css.map: HTTP error: status code 404, net::ERR_HTTP_RESPONSE_CODE_FAILURE
scripts.js?v=2.12.7:2068 Uncaught TypeError: Cannot read properties of null (reading 'style')
    at plugin_formcreator_changeValidators (scripts.js?v=2.12.7:2068:50)
    at HTMLSelectElement.onchange (form.form.php:1:1)
    at Object.trigger (base.min.js?v=9.5.1:25:71756)
    at HTMLSelectElement.<anonymous> (base.min.js?v=9.5.1:25:72337)
    at Function.each (base.min.js?v=9.5.1:14:3048)
    at jQuery.fn.init.each (base.min.js?v=9.5.1:14:1420)
    at jQuery.fn.init.trigger (base.min.js?v=9.5.1:25:72312)
    at i.define.n.select (base.min.js?v=9.5.1:446:34708)
    at s.<anonymous> (base.min.js?v=9.5.1:446:35552)
    at i.define.n.invoke (base.min.js?v=9.5.1:446:4990)

**Comment by izaacmenezes** [comment]
![image](https://user-images.githubusercontent.com/72463556/210886663-49a4964c-416f-4e30-8113-15fefa830f2c.png)

**Comment by izaacmenezes** [comment]
no meu caso funcionou usando a versão 2.11.2

**Comment by btry** [comment]
Hi @izaacmenezes 

Your errors are not the same as @tavoguardp . They seem completely unrelated to a button display problem.

Also, Formcreator 2.12 reached end of life. You should upgrade to GLPI 10 and Formcreator 2.13 . I cannot continue to provide community support for Formcrerator 2.12
